packages/core/src/command/consolidated/search/fhir-resource/search-consolidated.ts

```diff
@@ -129,26 +131,42 @@ async function searchFhirResources({
   });
 }
 
+/**
+ * Hydrates the missing references in the resources.
+ *
+ * Doesn't hydrate the Patient resource in the resources array.
+ *
+ * @param cxId - The customer ID.
+ * @param patientId - The patient ID.
+ * @param resources - The resources to hydrate.
+ * @param iteration - The iteration of the hydration.
+ * @param hydratePatient - Whether to hydrate the Patient resource. Optional, defaults to false.
+ * @returns The hydrated resources.
+ */
 export async function hydrateMissingReferences({
   cxId,
   patientId,
   resources,
   iteration = 1,
+  hydratePatient = false,
 }: {
   cxId: string;
   patientId: string;
   resources: Resource[];
   iteration?: number;
+  hydratePatient?: boolean;
 }): Promise<Resource[]> {
   const { log } = out("OS.hydrateMissingReferences");
 
+  if (iteration > maxHydrationAttempts) return resources;
+
   const { missingReferences } = getReferencesFromResources({ resources });
   const missingRefIds = missingReferences.flatMap(r => {
     const referenceId = r.id;
-    if (!referenceId || referenceId === patientId) return [];
+    if (!referenceId || (!hydratePatient && referenceId === patientId)) return [];
     return getEntryId(cxId, patientId, referenceId);
   });
-  if (missingRefIds.length < 1 || iteration >= maxHydrationAttempts) return resources;
+  if (missingRefIds.length < 1) return resources;
```

Comment:
hypernit; imo prefer === 0 over < 1 when possible.

Reply:
Defensive programming, code in a way that even w/ smaller bugs the code keeps working. I don't expect this from array.length, but it's a programming style... :)